### PR TITLE
add text and rounded rectangle via methods and dir change 

### PR DESCRIPTION
**Pull request by TG-Techie**
added four things:

- round_rect, a normal rect with rounded corners, 
- fill_round_rect, filled version of above
- text method (with an overwrite in __init__) 
- TGFONT01.py with a dict of data that defines the shape of the letters and characters 

and also some __init__ acceleratable things

### CODE REVIEW

**Comment by sommersoft** [comment]
@TG-Techie took me a few to understand the structure and proposed usage of this. Its a work in progress, so doesn't need to be fully polished yet, and may void my comments anyway. That said, I have a few open-ended review comments:

- **Standard Font(s)**:
  - If a standard font, or fonts, are going to be included, then the files need to be setup as a package. i.e.: create a folder `./adafruit_gfx/`, and move `adafruit_gfx.py` and any font files to that folder.
  - If `TGFONT01.py` is going to be _the_ standard font, I would change the name to something like `gfx_standard_font`. Nothing wrong with attributing the work to you, but that is what the license info in the file is for. (Please do not misunderstand this comment; I/we appreciate your contribution!!!)
  - If font objects can be passed into `__init__`, do all assignment/import there. This will help with memory usage, since currently the `std_font` will get imported even if you pass in a separate font object. So, instead of `try: import <module> as std_font` at the top, move it into `__init__` like so (shortened for brevity):

```python
def __init__(font=None):
    if not font:
        # use the standard font, which is part of the package.
        # since its part of the package, we don't need a try
        import <module> as std_font
    else:
        std_font = font
    # handle any type/instance checking
    self._font = std_font
```

As this took a little longer than I expected, I'll have to finish up my review/comments later. Off for family vacationy stuff...

File: .travis.yml [hunk 1]
```diff
@@ -27,7 +27,8 @@ install:
   - pip install --force-reinstall pylint==1.9.2
 
 script:
-  - pylint adafruit_gfx.py
+  - pylint adafruit_gfx/*.py
+  - pylint adafruit_gfx/fonts/*.py
```
Comment (kattni) [inline .travis.yml]: Perfect! Thank you for adding these lines to this file!

Comment (TG-Techie) [reply .travis.yml]: my pleasure

**Comment by TG-Techie** [comment]
screen shot 2018-12-26 at 7 23 21 am
 I have attached a photo of the raw hand typed data bits to create the letters. the 8th bit is always one and should be ignored since the height is 7. I feel like I should note that this is the 3rd to 4th time I've implemented text support and I find my method (on not intended) has become more condensed each time and has traded followability for speed.
thanks for the help, I'll start on the improvements. :-) . @sommersoft  hope your vacay time with family went well.

**Comment by TG-Techie** [comment]
and some example code:
`out = gfx.GFX(240,320,disp.pixel, disp.hline, disp.vline, disp.rect)


white = color(255,255,255)
blue = color(0,0,255)


out.fill_rect(0,0,100,50,io.green)
out.fill_round_rect(0,0,100,50,25,io.red)
out.round_rect(0,0,100,50,25,io.blue)
#out.fill_round_rect(0,0,100,50,20,io.blue)
#io.round_rect(0,0,100,50,20,io.red)
out.fill_circle(25,25,25,io.white)
out.hline(0,25,50,io.red)

#out._place_char(0,0,'A',1,white)
out.text(0,0,"""HELLO
WORLD __DEGREESIGN__
HELLO
WOvLD
HELLO
WORLD
""", 2,blue)
out.text(0,0,"""HELLO
WORLD __DEGREESIGN__
HELLO
WOvLD
HELLO
WORLD
""", 1,white)
time.sleep(10)
out.fill_round_rect(0,0,239,315,15,color(255,255,128))
out.round_rect(0,0,239,315,15,color(255,0,255))
#for i in range(7):
    #out.round_rect(50+i,200+i,50-i*2,70-i*2,20-i,color(255,0,255))
import time, random as rand

while 1:
    time.sleep(.1)
    todo = (out.fill_round_rect, out.round_rect)[rand.randint(0,1)]
    x = rand.randint(0,100)
    y = rand.randint(0,150)
    width = min(rand.randint(20,200),240-x-1)
    height = min(rand.randint(20,300),320-y-1)
    r = rand.randint(10,70)
    todo(x,y,width,height,r,color(128+rand.randint(0,127),128+rand.randint(0,127),128+rand.randint(0,127)))` where disp is a display one has setup and color is rgb, in this case.

**Comment by TG-Techie** [comment]
WOOO HOOO

**Comment by dherrada** [comment]
@sommersoft were these changes the ones you wanted?

**Comment by kattni** [comment]
@makermelissa Please give this one more review and feel free to merge it as you see fit.

**Comment by siddacious** [comment]
@kattni , @makermelissa, @TG-Techie  I pooped on this PR by merging the Actions update underneath it. `.travis.yml` should be removed as in the Actions PR. I can have Dylan put in a PR against the incoming branch unless @TG-Techie beats him to it. I'm open to other options as well 👂

**Comment by makermelissa** [comment]
Can you fix and push to TG-Techie's repo?

**Comment by makermelissa** [comment]
Make a separate branch first so you don't mess that up.

**Comment by makermelissa** [comment]
@siddacious I just cleaned up your mess. :) Easy enough.

**Comment by siddacious** [comment]
My name is Bryan and I approve this cleanup. Good to merge from my Actions-oriented POV

Thank you @makermelissa

**Pull request merged**